src/probnum/filtsmooth/gaussfiltsmooth/kalmanposterior.py

```diff
@@ -11,8 +13,9 @@
 from probnum.filtsmooth.filtsmoothposterior import FiltSmoothPosterior
 
 
-class KalmanPosterior(FiltSmoothPosterior):
-    """Posterior Distribution after (Extended/Unscented) Kalman Filtering/Smoothing.
+class KalmanPosterior(FiltSmoothPosterior, abc.ABC):
+    """Interface for posterior distribution after (extended/unscented) Kalman
+    filtering/smoothing.
```

Comment:
filtering and smoothing posterior share almost anything, so they have this interface (it is abc!)

The KalmanODESolution for example only sees a `KalmanPosterior` object, and is not aware whether it is a "FilteringODESolution" or a "SmoothingODESolution".

Reply:
That's quite cool :)